### PR TITLE
whole nav buttons clickable + scroll

### PR DESCRIPTION
**Pull request by vict0rsch**
* moved style from `li` to `a` so that the whole nav box is clickable not just the text within it
* added `custom.js` with a few lines to smoothly scroll to the first challenge of selected category
  * scroll only if first challenge not visible to user
  * scroll time depends on distance to element (but >300 and <800 ms)

### CODE REVIEW

File: javascripts/custom.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if (cat === "tous") return;
+
+        const el = $(`.gallery .${cat}`).first();
+        const h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
+        const p = el.position();
+
+        if (p.top > h) {
+
+            let scrollTime = Math.round((p.top - h) / 3);
+            scrollTime = Math.min(scrollTime, 800);
```
Comment (AntoineAugusti) [inline javascripts/custom.js]: Swap min and max

Comment (vict0rsch) [reply javascripts/custom.js]: what do you mean? you don't want more than 800 ms and not less than 300

Comment (AntoineAugusti) [reply javascripts/custom.js]: Sorry, read that too fast 🤦‍♂ After testing it, 300ms seems a bit fast

**Comment by AntoineAugusti** [comment]
Démo : https://deploy-preview-199--entrepreneur-interet-general.netlify.com/defis.html

File: javascripts/custom.js (hunk 1 above)
Comment (vict0rsch) [reply javascripts/custom.js]: I'll let you decide of a decent minimal time :)

**Comment by AntoineAugusti** [comment]
@jbledevehat What do you think of this?

**Comment by jbledevehat** [comment]
Très bien... Pourrions-nous ajouter un bouton de retour en haut de page par la même occasion ?

**Comment by vict0rsch** [comment]
2e PR non?

**Comment by vict0rsch** [comment]
@jbledevehat  @AntoineAugusti added back to top button, I'll let you decide on style if you're not happy with it :)

**Comment by AntoineAugusti** [comment]
Thanks a lot @vict0rsch 🙏

**Pull request merged**